src/components/application_manager/src/state_controller_impl.cc

```diff
@@ -955,6 +957,15 @@ void StateControllerImpl::OnStateChanged(ApplicationSharedPtr app,
 
   if (new_state->hmi_level() == mobile_apis::HMILevel::HMI_NONE) {
     app->ResetDataInNone();
+    if (mobile_apis::HMILevel::INVALID_ENUM == old_state->hmi_level()) {
```

Comment:
Can you please explain when this code will run when `GetDefaultHmiLevel` returns a status other than `NONE`?

Reply:
@iCollin I believe this code will not be executed if app transits from INVALID to whatever except NONE. It is a good point and I think this block should be moved to the end of function

Reply:
@iCollin I believe this code will not be executed if app transits from INVALID to whatever except NONE. It is a good point and I think this block should be moved to the end of function

Reply:
@AKalinich-Luxoft At the end of this function i suggest this is the implemented fix:

```
  if (mobile_apis::HMILevel::INVALID_ENUM == old_state->hmi_level() && app_mngr_.GetDefaultHmiLevel(app) == new_state->hmi_level()) {
    const uint32_t app_id = app->app_id();
    const uint32_t corr_id =
        postponed_activation_controller_.GetPendingActivationCorrId(app_id);
    if (corr_id > 0) {
      app_mngr_.GetPolicyHandler().OnActivateApp(app_id, corr_id);
      postponed_activation_controller_.RemoveAppToActivate(app_id);
    }
  }
```

Reply:
This should address the possibility for a default hmi level other than NONE

Reply:
@iCollin @JackLivio fixed in https://github.com/smartdevicelink/sdl_core/pull/3557/commits/4605e9859af606ef434484e8c9ea094918e4d365